src/profile/directory.mo

```diff
@@ -30,7 +34,19 @@ module {
       ignore hashMap.set(userId, profile);
     };
 
-    public func search (term : Text) : [Profile] {
+    public func findOne (userId : PrincipalId) : ?Profile {
+      hashMap.get(userId)
+    };
+
+    public func findMany (userIds : [PrincipalId]) : [Profile] {
```

Comment:
What's the desired behavior and result for when some of these are not found?

Reply:
Right now it would trap, right? I think I'd rather return an array of only the found ones.

Reply:
Yes, it will trap.

Ok, in that case, if you use an `Array.mapFilter` in place of the step where you `Array.map`, I think you can get exactly what you want.

Reply:
Oh, looks like we have `Array.filter` and `Array.map`; we need to combine those, I think.

Just a sec.

Reply:
One bummer about `map` and `filter` being class methods instead of instance methods is that I can't do: `someArray.map(operation).filter(operation)` instead of `Array.filter(Array.map(someArray, operation), operation)`. I suppose I should look at `Function.compose`.